### DISCUSSION TITLE
Register handlers

### DISCUSSION THREAD

**Discussion by parsa01** [LuckyPennySoftware/MediatR#891] [General]
HI @jbogard
I'm just curious why did you use this `addIfAlreadyExists` in the `ConnectImplementationsToTypesClosing` register handlers function? 
[Service Register](https://github.com/jbogard/MediatR/blob/master/src/MediatR/Registration/ServiceRegistrar.cs)

And why did you try to register handlers even if they did and why didn't you do it for the others?

**Comment by jbogard** [comment]
I can't remember why this was there. I'm re-looking at all this scanning stuff actually, it's a big source of confusion and bugs.

**Comment by parsa01** [comment]
Oh okay, Thanks for the reply
